runtime/src/main/java/com/dylibso/chicory/runtime/Module.java

```diff
@@ -200,9 +202,17 @@ public Instance instantiate(HostFunction[] hostFunctions) {
         }
 
         var globalImportsOffset = 0;
+        var functionImportsOffset = 0;
         for (int i = 0; i < imports.length; i++) {
-            if (imports[i].getDesc().getType() == ImportDescType.GlobalIdx) {
-                globalImportsOffset++;
+            switch (imports[i].getDesc().getType()) {
```

Comment:
let's see how this goes, but I'll probably move this code to the `Instance` constructor next time I touch it.